### ISSUE TITLE
🎯 Initial Project Setup Tasks

### ISSUE THREAD

**Issue by roninjin10** [evmts/onchain-clippy#1]
# 🎯 Initial Project Setup Tasks

These tasks set up the basic boilerplate for this project. Up to 2500 AI16z tokens are up for grabs

### Repository Setup
- [x] Initialize basic repository structure
- [x]  [s29papi](https://github.com/s29papi) Initialize bun project with TypeScript configuration (100 AI16z) 
- [x] [Larhra](https://github.com/Larhra) Add .gitignore for bun/node (100 AI16z)
- [x] khazra](https://github.com/khazra) Set up Nix development environment (100 AI16z)
- [x] Create basic README.md with project overview and goals

### Documentation
- [x] dd CODE_OF_CONDUCT.md (because Clippy believes in ethical coding) (100 AI16z) [Houseofkholin](https://github.com/Houseofkholin)
- [x] Create CONTRIBUTING.md guidelines in README
- [x] Add LICENSE file (MIT)

### CI/CD

These tasks are blocked by other tasks being done first

- [x]  [hidd3n19](https://github.com/hidd3n19) Set up GitHub Actions for: (200 AI16z)
  - [x] Lint checks
  - [x] Type checking
  - [x] Test running
  - [x] Docker build verification
  (200 AI16z for complete CI setup. See tevm monorepo for guidance on how to do this)
- [x] ⏳ [hidd3n19](https://github.com/hidd3n19) Add status badges to README (50 AI16z)

### Development Environment
- [x] [s29papi](https://github.com/s29papi) Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z) 
- [x] Add Biome configuration (following tevm-monorepo) (50 AI16z) [Larhra](https://github.com/Larhra)
- [x] Add a hello world chrome extension with docs on how to set up dev env to use it in CONTRIBUTING.md (200 AI16z) [g4titanx](https://github.com/g4titanx)
- [ ] ⏳ Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension (400 AI16z) [khazra](https://github.com/khazra)
- [ ] Add initial lore and json configuration to CLIPPY eliza (100 AI16z)
- [x] Set up basic Docker development environment

### Core Infrastructure
- [x] Set up basic TEE implementation structure
- [x] Initialize light client integration
- [x] Add test demonstrating light client running in Docker
- [ ] Add WhatsABI integration to decode contracts and verify contract status in the chrome extension (100 AI16z)
- [x] #7

## Requirements

- Follow code style from [tevm-monorepo](https://github.com/evmts/tevm-monorepo)
- Must include solana address for gifted token distribution
- Keep commits atomic and well-documented
- Each PR should focus on a single task
- Comment on this issue to claim a task before starting work

## Notes

Remember, Clippy is watching your code... and he only wants to help! 👀
All gifted payouts are discretionary
Any non listed contributions may also receive gifted AI16z as a thank you at our discression
If you have contribution not listed that you think would be helpful feel free to propose it

**Comment by khazra** [comment]
Hi! I'd love to contribute. I can take anything/everything. You don't want any single contributor taking dibs on the entire issue, right?

**Comment by s29papi** [comment]
Hi, I want to get assigned

**Comment by roninjin10** [comment]
> Hi! I'd love to contribute. I can take anything/everything. You don't want any single contributor taking dibs on the entire issue, right?

Ideally only one or 2 sub issues at a time so others can participate too. Any issues look specifically interesting? I can help get you started.

**Comment by roninjin10** [comment]
> Hi, I want to get assigned

Which sub issues do you want to take?

**Comment by s29papi** [comment]
@roninjin10 

1. Task: Initialize bun project with TypeScript configuration (100 AI16z) 
2. Task: Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z)

**Comment by Larhra** [comment]
would start of with the task: Add .gitignore for bun/node (100 AI16z)

**Comment by roninjin10** [comment]
> @roninjin10
> 
> 1. Task: Initialize bun project with TypeScript configuration (100 AI16z)
> 2. Task: Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z)

All yours. Just to be clear we will be using bun for runtime and tests but pnpm for monorepo setup. Tevm monorepo is a good example to follow of how to do this. 

- [ ] Set up the monorepo with a app/extension package and a packages/light-client package. 
- [ ] The app/extension package should be importing something from the packages/light-client package

**Comment by khazra** [comment]
> > Hi! I'd love to contribute. I can take anything/everything. You don't want any single contributor taking dibs on the entire issue, right?
> 
> Ideally only one or 2 sub issues at a time so others can participate too. Any issues look specifically interesting? I can help get you started.

I could start with the `Set up project structure following tevm-monorepo conventions` and `Set up Nix development environment`

**Comment by roninjin10** [comment]
> would start of with the task: Add .gitignore for bun/node (100 AI16z)

All yours @Larhra . Since this is a monorepo make sure the git ignore ignores stuff in subfolders too

**Comment by roninjin10** [comment]
> > > Hi! I'd love to contribute. I can take anything/everything. You don't want any single contributor taking dibs on the entire issue, right?
> > 
> > 
> > Ideally only one or 2 sub issues at a time so others can participate too. Any issues look specifically interesting? I can help get you started.
> 
> I could start with the `Set up project structure following tevm-monorepo conventions` and `Set up Nix development environment`

tevm-monorepo is taken but Nix is all yours. Tevm monorepo has a nix setup already. Please add instructions for nix to CONTRIBUTING.md too

**Comment by Houseofkholin** [comment]
@roninjin10 i will take code of conduct and github actions

**Comment by g4titanx** [comment]
@roninjin10 i will take on "Add a hello world chrome extension with docs on how to set up dev env..." and if it's open, i'd also take on "Set up GitHub Actions..."

**Comment by hidd3n19** [comment]
@roninjin10

1. Task: Set up GitHub Actions for: (200 AI16z)
2. Task: Add status badges to README (50 AI16z)

**Comment by Houseofkholin** [comment]
@roninjin10 what's the initial lore ?

**Comment by Larhra** [comment]
@roninjin10 

i'll add Add Biome configuration (following tevm-monorepo) (50 AI16z). To my initial .gitignore task

**Comment by roninjin10** [comment]
> @roninjin10 what's the initial lore ?

Some of it is the readme. We need to expand on it and create an entire eliza json config

**Comment by roninjin10** [comment]
> @roninjin10 i will take code of conduct and github actions

Code of conduct is all yours

Let's hold off on github actions until we have enough other stuff added to be able to lint or test something

**Comment by g4titanx** [comment]
> @roninjin10 i will take on "Add a hello world chrome extension with docs on how to set up dev env..." and if it's open, i'd also take on "Set up GitHub Actions..."

hey @roninjin10 since we are gonna be pausing on github actions i'll simply take on "Add a hello world chrome extension with docs on how to set up dev env to use it in CONTRIBUTING.md (200 AI16z)"

**Comment by roninjin10** [comment]
> > @roninjin10 i will take on "Add a hello world chrome extension with docs on how to set up dev env..." and if it's open, i'd also take on "Set up GitHub Actions..."
> 
> hey @roninjin10 since we are gonna be pausing on github actions i'll simply take on "Add a hello world chrome extension with docs on how to set up dev env to use it in CONTRIBUTING.md (200 AI16z)"

Sounds good. You will want to branch off of https://github.com/evmts/onchain-clippy/pull/3 for this pr since it sets up the folder for the chrome extension. Or you can simply rebase onto it after it merges

**Comment by roninjin10** [comment]
> @roninjin10
> 
> i'll add Add Biome configuration (following tevm-monorepo) (50 AI16z). To my initial .gitignore task

All yours

**Comment by Houseofkholin** [comment]
@roninjin10 can I take over the lore ?

**Comment by s29papi** [comment]
@roninjin10 

Could you give details on this, abit confused Intercept eth_sendRawTransaction calls with extension and simulate with Tevm (500 AI16z). A hint would do.

**Comment by hidd3n19** [comment]
I did like to do this then, 

Add WhatsABI integration to decode contracts and verify contract status in the chrome extension (100 AI16z)

Add status badges to README (50 AI16z)

**Comment by roninjin10** [comment]
@s29papi I added more context to a new issue https://github.com/evmts/onchain-clippy/issues/7 . This is currently blocked by me reviewing and merging https://github.com/evmts/onchain-clippy/pull/6

**Comment by roninjin10** [comment]
> I did like to do this then,
> 
> Add WhatsABI integration to decode contracts and verify contract status in the chrome extension (100 AI16z)
> 
> Add status badges to README (50 AI16z)

WhatsABI is currently blocked by #6 first and status badges is blocked by the CI/CD one. Do you want ot add CI/CD which is mostly just copying the CI/CD from Tevm monorepo?

**Comment by roninjin10** [comment]
> @roninjin10 can I take over the lore ?

Yea for this you need to make a new package in packages/agents and make a AI16z eliza character using the json format https://github.com/ai16z/eliza

**Comment by roninjin10** [comment]
Merged some prs to unblock. Will verify them and send out gifts tomorrow

**Comment by hidd3n19** [comment]
@roninjin10 
I will do the CI/CD one and the status badges one, too.
I would appreciate it if they were assigned to me

**Comment by khazra** [comment]
@roninjin10 
I'd like to take on `Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension` and `Add WhatsABI integration to decode contracts and verify contract status in the chrome extension`.

**Comment by g4titanx** [comment]
hey @roninjin10 i would love to take on " Set up GitHub Actions..." as soon as it's ready

**Comment by roninjin10** [comment]
> @roninjin10 I'd like to take on `Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension` and `Add WhatsABI integration to decode contracts and verify contract status in the chrome extension`.

Those are both bigger tasks. What about just starting with the whatsabi inegration?

**Comment by roninjin10** [comment]
> @roninjin10 I will do the CI/CD one and the status badges one, too. I would appreciate it if they were assigned to me

All yours

**Comment by khazra** [comment]
> > @roninjin10 I'd like to take on `Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension` and `Add WhatsABI integration to decode contracts and verify contract status in the chrome extension`.
> 
> Those are both bigger tasks. What about just starting with the whatsabi inegration?

Sure, I'm on it.

**Comment by roninjin10** [comment]
> Merged some prs to unblock. Will verify them and send out gifts tomorrow

Will send out gifts tomorrow. Wallet I was sending from doesn't have Sol in it will fix that tomorrow

**Comment by Houseofkholin** [comment]
> > Merged some prs to unblock. Will verify them and send out gifts tomorrow
> 
> Will send out gifts tomorrow. Wallet I was sending from doesn't have Sol in it will fix that tomorrow

Just wanted to confirm has it been sent ?

**Comment by s29papi** [comment]
@roninjin10 is this still active ?

**Comment by roninjin10** [comment]
Still active will circle back tomorrow

**Comment by g4titanx** [comment]
hey @roninjin10 have you sent out gifts?

**Comment by khazra** [comment]
Hi, @roninjin10, all the best in 2025!

Considering that you've never assigned me to the `whatsabi` task and that @s29papi has already started to integrate it as part of #9, I'd like to take on the `Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension (400 AI16z)`.

**Comment by s29papi** [comment]
Hi @roninjin10 everyone here I guess is on the lookout for this to get active back, and also for collecting their token gifts. This would really go a long way, for each dev I guess.

**Comment by Larhra** [comment]
Hello @roninjin10 when would gifts be sent out, also just want to ask is this still in development ?

**Comment by roninjin10** [comment]
Hope everybody had a great holiday! I went through and updated all the tasks. The most important task we need to get done is  `Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension (400 AI16z)` and then I can make a ticket for the next round of work

**Comment by s29papi** [comment]
@roninjin10 gm. Thanks for the payout: please could you review some task done by me 
1. Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z) was done as a single task with  Initialize bun project with TypeScript configuration (100 AI16z)  https://github.com/evmts/onchain-clippy/pull/3

2. whatsabi wasn't assinged but i had to do it to get my task done.

Appreciate irrespective 😊

**Comment by s29papi** [comment]
> @roninjin10 gm. Thanks for the payout: please could you review some task done by me
> 
> 1. Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z) was done as a single task with  Initialize bun project with TypeScript configuration (100 AI16z)  [Initialized bun project with TypeScript configuration and pnpm monorepo setup (awaiting review for completion) #3](https://github.com/evmts/onchain-clippy/pull/3)
> 2. whatsabi wasn't assinged but i had to do it to get my task done.
> 
> Appreciate irrespective 😊

@roninjin10 on this I should have created a separate pull request one to initialize bun and the other to setup a pnpm typescript. but i had it as a single pr as this was to start the project.

on 2. for whatsabi i know your rule on assigning, but it was done so I could get the abi. 

Would really appreciate, if you can look into this.

**Comment by khazra** [comment]
> Hope everybody had a great holiday! I went through and updated all the tasks. The most important task we need to get done is  `Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension (400 AI16z)` and then I can make a ticket for the next round of work

I've asked to be assigned for this a couple times already :) I'm happy to do it.

I think the task is rather clear, but if you want anything specific regarding the implementation and/or UX, please let me know.

**Comment by g4titanx** [comment]
hey @roninjin10 seen the gifts, thanks
i'd be glad to take on more tasks

**Comment by malik672** [comment]
no more tasks for you @g4titanx

**Comment by roninjin10** [comment]
> > Hope everybody had a great holiday! I went through and updated all the tasks. The most important task we need to get done is  `Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension (400 AI16z)` and then I can make a ticket for the next round of work
> 
> I've asked to be assigned for this a couple times already :) I'm happy to do it.
> 
> I think the task is rather clear, but if you want anything specific regarding the implementation and/or UX, please let me know.

All yours

**Comment by roninjin10** [comment]
> [@roninjin10](https://github.com/roninjin10) what's the initial lore ?

We still need someone to make the initial lore if you want to pick that one up

**Comment by s29papi** [comment]
@roninjin10 i could do the lore if you don't mind

**Comment by s29papi** [comment]
> @roninjin10 gm. Thanks for the payout: please could you review some task done by me 
> 
> 1. Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z) was done as a single task with  Initialize bun project with TypeScript configuration (100 AI16z)  https://github.com/evmts/onchain-clippy/pull/3
> 
> 
> 
> 2. whatsabi wasn't assinged but i had to do it to get my task done.
> 
> 
> 
> Appreciate irrespective 😊

@roninjin10 please would appreciate if this is reviewed

**Comment by s29papi** [comment]
@roninjin10 configuring typescript pnpm monorepo was assigned to me as well as setting up a bun project, I did both together in a single pr, this prevented me from getting payment on typescript monorepo and only getting payment on bun. This can be reviewed in the tasks assigned above. It was a mistake on my end. 

My sol address: 

8fFsLKGv5sizL97mjtrVvPu9trQAHZx5HBeLGzLVAt2o

Would appreciate for this to be considered

**Comment by s29papi** [comment]
@roninjin10 I wasn't assigned whatsabi task, I only went ahead to do it because it helped me accomplish my own task, I should have clarified and awaited it to be assigned. 

My sol address: 

8fFsLKGv5sizL97mjtrVvPu9trQAHZx5HBeLGzLVAt2o

Would appreciate for this to be considered

**Comment by roninjin10** [comment]
Hey @s29papi apologies for that oversight. I won't be able to gift again til next week though

**Comment by roninjin10** [comment]
Hey ya'll, I'm thinking about starting a telegram so informal discussion can happen and we can keep this project moving forward faster. What do you think?

**Comment by roninjin10** [comment]
Hey ya'll, I'm thinking about starting a telegram so informal discussion can happen and we can keep this project moving forward faster. What do you think?

@s29papi @malik672 @g4titanx @khazra @Larhra @Houseofkholin @hidd3n19

**Comment by s29papi** [comment]
> > [@roninjin10](https://github.com/roninjin10)
> > 
> > 1. Task: Initialize bun project with TypeScript configuration (100 AI16z)
> > 2. Task: Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z)
> 
> All yours. Just to be clear we will be using bun for runtime and tests but pnpm for monorepo setup. Tevm monorepo is a good example to follow of how to do this.
> 
> * [ ]  Set up the monorepo with a app/extension package and a packages/light-client package.[ ]  The app/extension package should be importing something from the packages/light-client package

@roninjin10 I completed task 2 assigned. But yet to receive the payout. (200 AI16z)
@roninjin10 I also completed and resolved Whatsabi integration (a different task) (100 Ai16z)

8fFsLKGv5sizL97mjtrVvPu9trQAHZx5HBeLGzLVAt2o

**Comment by roninjin10** [comment]
> > > [@roninjin10](https://github.com/roninjin10)
> > > 
> > > 1. Task: Initialize bun project with TypeScript configuration (100 AI16z)
> > > 2. Task: Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z)
> > 
> > 
> > All yours. Just to be clear we will be using bun for runtime and tests but pnpm for monorepo setup. Tevm monorepo is a good example to follow of how to do this.
> > 
> > 
> > * [ ]   Set up the monorepo with a app/extension package and a packages/light-client package.[ ]  The app/extension package should be importing something from the packages/light-client package
> 
> [@roninjin10](https://github.com/roninjin10) I completed task 2 assigned. But yet to receive the payout. (200 AI16z) [@roninjin10](https://github.com/roninjin10) I also completed and resolved Whatsabi integration (a different task) (100 Ai16z)
> 
> 8fFsLKGv5sizL97mjtrVvPu9trQAHZx5HBeLGzLVAt2o

I can’t send any more gifts til monday but this seems not right. The whatsabi task doesn’t look like it was assigned so that would explain why that didn’t get sent but I already see 600 sent to this address

**Comment by Houseofkholin** [comment]
@roninjin10 any more task and yes i do agree a telegram group is good for this

**Comment by roninjin10** [comment]
New telegram https://t.me/+7a6QN3wuUL0zOGEx

**Comment by s29papi** [comment]
> > > > [@roninjin10](https://github.com/roninjin10)
> > > > 
> > > > 1. Task: Initialize bun project with TypeScript configuration (100 AI16z)
> > > > 2. Task: Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z)
> > > 
> > > 
> > > All yours. Just to be clear we will be using bun for runtime and tests but pnpm for monorepo setup. Tevm monorepo is a good example to follow of how to do this.
> > > 
> > > 
> > > * [ ]    Set up the monorepo with a app/extension package and a packages/light-client package.[ ]  The app/extension package should be importing something from the packages/light-client package
> > 
> > 
> > [@roninjin10](https://github.com/roninjin10) I completed task 2 assigned. But yet to receive the payout. (200 AI16z) [@roninjin10](https://github.com/roninjin10) I also completed and resolved Whatsabi integration (a different task) (100 Ai16z)
> > 8fFsLKGv5sizL97mjtrVvPu9trQAHZx5HBeLGzLVAt2o
> 
> I can’t send any more gifts til monday but this seems not right. The whatsabi task doesn’t look like it was assigned so that would explain why that didn’t get sent but I already see 600 sent to this address

Ok, but the task 2: Configure TypeScript pnpm monorepo (following tevm-monorepo for guidance) (200 AI16z)
is valid in this case because it was assigned, and not payed.

600 ai16z is valid for both task 1 and task 7, but task 2 was assigned me and wasn't payed:  ( Task 1: Initialize bun project with TypeScript configuration (100 AI16z) + Task 7: https://github.com/evmts/onchain-clippy/issues/7)

**Comment by roninjin10** [comment]
Ah yes I think there is a 200 missing along with the unassigned task. I’ll send for both but in future try to get assigned task before doing it. I will still send for it but we want to avoid race conditions of two people implementing same task

**Comment by khazra** [comment]
Hi @roninjin10 , feel free to unassign me from the 'Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension'. I won't be able to contribute anything until 15 Feb at best. I'm sure @g4titanx or @Larhra can handle the task sooner.

**Comment by Houseofkholin** [comment]
> Hi [@roninjin10](https://github.com/roninjin10) , feel free to unassign me from the 'Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension'. I won't be able to contribute anything until 15 Feb at best. I'm sure [@g4titanx](https://github.com/g4titanx) or [@Larhra](https://github.com/Larhra) can handle the task sooner.

@roninjin10 i would love to take this on

**Comment by g4titanx** [comment]
hey @roninjin10 

> Add initial lore and json configuration to CLIPPY eliza (100 AI16z)

i would love to take this on

**Comment by g4titanx** [comment]
> Hi [@roninjin10](https://github.com/roninjin10) , feel free to unassign me from the 'Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension'. I won't be able to contribute anything until 15 Feb at best. I'm sure [@g4titanx](https://github.com/g4titanx) or [@Larhra](https://github.com/Larhra) can handle the task sooner.

yes i have an experience with eliza and it wont be a problem

**Comment by Houseofkholin** [comment]
> > Hi [@roninjin10](https://github.com/roninjin10) , feel free to unassign me from the 'Add ELIZA integration in TEE with API for basic AI responses in a simple chatbox in chrome extension'. I won't be able to contribute anything until 15 Feb at best. I'm sure [@g4titanx](https://github.com/g4titanx) or [@Larhra](https://github.com/Larhra) can handle the task sooner.
> 
> yes i have an experience with eliza and it wont be a problem

yeah I came first :)

**Comment by s29papi** [comment]
> Ah yes I think there is a 200 missing along with the unassigned task. I’ll send for both but in future try to get assigned task before doing it. I will still send for it but we want to avoid race conditions of two people implementing same task

I appreciate this, here is my addy 8fFsLKGv5sizL97mjtrVvPu9trQAHZx5HBeLGzLVAt2o
